python-client/giskard/testing/tests/statistic.py

```diff
@@ -224,3 +226,36 @@ def test_disparate_impact(model: BaseModel, dataset: Dataset, protected_slicing_
         passed=bool((disparate_impact_score > min_threshold) * (disparate_impact_score < max_threshold)),
         messages=messages,
     )
+
+
+@test(name="Right Label", tags=["heuristic", "classification"])
+def test_independence_chi_square(model: BaseModel, dataset: Dataset,
+                                 slicing_function: SlicingFunction, threshold: float = 0.1) -> TestResult:
```

Comment:
`p_value < 0.1` is a implausibly high threshold! I would set it at maximum to 0.01. Generally Χ² is quite powerful, so we will not miss meaningul detections if the sample size is decent (for example I use  Χ² tests in `TextSlicer` to find deviant tokens with p_value < 1e-3).